lib/internal/modules/esm/loader.js

```diff
@@ -16,14 +16,20 @@ const {
 const { getOptionValue } = require('internal/options');
 const { pathToFileURL } = require('internal/url');
 const { emitExperimentalWarning } = require('internal/util');
+const { isPromise } = require('internal/util/types');
 const {
   getDefaultConditions,
 } = require('internal/modules/esm/utils');
 let defaultResolve, defaultLoad, importMetaInitializer;
 
-function newModuleMap() {
-  const ModuleMap = require('internal/modules/esm/module_map');
-  return new ModuleMap();
+function newModuleResolveMap() {
+  const { ModuleResolveMap } = require('internal/modules/esm/module_map');
+  return new ModuleResolveMap();
+}
+
+function newModuleLoadMap() {
+  const { ModuleLoadMap } = require('internal/modules/esm/module_map');
+  return new ModuleLoadMap();
 }
```

Comment:
We call them in the tests, and yes also when we have several loaders (although now I don't think it ever happens on the same thread, so that last one would not be a concern).